ethpm_cli/config.py

```diff
@@ -29,3 +38,43 @@ def __init__(self, args: Namespace) -> None:
             if not self.ethpm_dir.is_dir():
                 self.ethpm_dir.mkdir()
         validate_ethpm_dir(self.ethpm_dir)
+
+        # Setup xdg ethpm dir
+        xdg_ethpm_dir = get_xdg_ethpmcli_root()
+        w3 = Web3(load_provider_from_uri(INFURA_HTTP_URI))
+        if xdg_ethpm_dir.is_dir():
+            validate_xdg_ethpm_dir(xdg_ethpm_dir, w3)
+        else:
+            initialize_xdg_ethpm_dir(xdg_ethpm_dir, w3)
+
+
+def validate_xdg_ethpm_dir(xdg_dir, w3):
+    chain_data_path = xdg_dir / IPFS_CHAIN_DATA
+    validate_chain_data_store(chain_data_path, w3)
+    if not (xdg_dir / KEYFILE_PATH).is_file():
+        raise ValidationError(
+            f"Invalid xdg ethpm dir found @ {xdg_dir}. No keyfile found."
+        )
+
+
+def initialize_xdg_ethpm_dir(ethpm_dir: Path, w3: Web3) -> None:
+    ethpm_dir.mkdir()
+    os.environ['XDG_ETHPMCLI_ROOT'] = str(ethpm_dir)
+    chain_data_path = ethpm_dir / IPFS_CHAIN_DATA
+    chain_data_path.touch()
+    init_json = {
+        "chain_id": w3.eth.chainId,
+        "scraped_blocks": [{"min": "0", "max": "0"}],
+    }
+    write_updated_chain_data(chain_data_path, init_json)
+    ethpm_keyfile = ethpm_dir / KEYFILE_PATH
+    ethpm_keyfile.touch()
+
+
+def write_updated_chain_data(
+    chain_data_path: Path, updated_data: Dict[str, Any]
+) -> None:
+    tmp_pkg_dir = Path(tempfile.mkdtemp())
+    tmp_data = tmp_pkg_dir / "chain_data.json"
```

Comment:
I realize you could encapsulate this into a utility like this.

```python
with atomic_replace(Path("some/path.txt")) as file:
    file.write('some text')
    # here nothing has actually been done to the original file
# here the context manager has replaced the original file
```

Seems like a simple and nice utility to do this concisely